src/types/onyx/Policy.ts

```diff
@@ -68,18 +69,101 @@ type TaxRatesWithDefault = {
     taxes: TaxRates;
 };
 
-// These types are for the Integration connections for a policy (eg. Quickbooks, Xero, etc).
-// This data is not yet used in the codebase which is why it is given a very generic type, but the data is being put into Onyx for future use.
-// Once the data is being used, these types should be defined appropriately.
-type ConnectionLastSync = Record<string, unknown>;
-type ConnectionData = Record<string, unknown>;
-type ConnectionConfig = Record<string, unknown>;
+type ConnectionLastSync = {
+    successfulDate?: string;
+    errorDate?: string;
+    isSuccessful: boolean;
+    source: 'DIRECT' | 'EXPENSIFYWEB' | 'EXPENSIFYAPI' | 'AUTOSYNC' | 'AUTOAPPROVE';
+};
+
+type Account = {
+    glCode?: string;
+    name: string;
+    currency: string;
+    id: number;
+};
+
+type Employee = {
+    id: number;
+    firstName: string;
+    lastName: string;
+    name: string;
+    email: string;
+};
+
+type Vendor = {
+    id: number;
+    name: string;
+    currency: string;
+    email: string;
+};
+
+type ConnectionData = {
+    country: string;
+    edition: string;
+    homeCurrency: string;
+    isMultiCurrencyEnabled: boolean;
+
+    journalEntryAccounts: Account[];
+    bankAccounts: Account[];
+    creditCards: Account[];
+    accountsReceivable: Account[];
+    accountsPayable: Account[];
+    otherCurrentAssetAccounts: Account[];
+
+    taxCodes: string[];
+    employees: Employee[];
+    vendors: Vendor[];
+};
+
+type IntegrationEntity = 'NONE' | 'DEFAULT' | 'TAG' | 'REPORT_FIELD';
```

Comment:
This name is pretty confusing, and represents how we map a certain type of data from the integration to something in the policy. Maybe something like `IntegrationEntityMappingType` would be clearer?

Reply:
`IntegrationEntityMappingType` was also my original choice, but there is [a specification](https://github.com/Expensify/App/blob/main/contributingGuides/TS_STYLE.md#naming-conventions:~:text=Do%20not%20postfix%20type%20aliases%20with%20Type.) in the TypeScript style guide that we should avoid post-fixing type names with `Type`.

@francoisl I'll change this to `IntegrationEntityMapping`.